src/psyclone/dynamo0p3.py

```diff
@@ -1295,7 +1301,10 @@ def initialise(self, parent):
             # from the field proxy and undf is not required.
             if not (self._dofs_only and Config.get().distributed_memory):
                 if self._invoke.field_on_space(function_space):
-                    undf_name = function_space.undf_name
+                    if self._invoke.operates_on_dofs_only:
```

Comment:
Again, I'm not convinced that this change is needed. I guess it comes back to why we need a 'bare' undf_name?